src/reducers/join.ts

```diff
@@ -0,0 +1,54 @@
+import _ from 'lodash'
+import existingThoughtChange from './existingThoughtChange'
+import { getAllChildren, getChildPath, getNextRank, getPrevRank } from '../selectors'
+import { SimplePath } from '../types'
+import { head, parentOf, pathToContext, reducerFlow } from '../util'
+import { State } from '../util/initialState'
+import existingThoughtMove from './existingThoughtMove'
+import existingThoughtDelete from './existingThoughtDelete'
+
+/** Join two or more thoughts split by spaces. */
+const join = (state: State) => {
+
+  const { cursor } = state
+
+  if (!cursor) return state
+
+  const contextRanked = pathToContext(parentOf(cursor))
+  const contextChildren = getAllChildren(state, contextRanked)
+  const { value, rank } = head(cursor)
+  const siblings = contextChildren.filter(child => child.value !== value && child.rank !== rank)
+
+  const actions = siblings.reduce((acc, sibling, index) => {
+
+    const pathToSibling = [...parentOf(cursor), sibling]
+    const children = getAllChildren(state, pathToContext(pathToSibling))
+
+    return [...acc, ...children.map(child => {
+      const oldPath = getChildPath(state, child, pathToSibling as SimplePath)
```

Comment:
`pathToSibling` has to be properly converted to a `SimplePath` here. Casting is only appropriate if the `Path` is guaranteed to be a `SimplePath`, such as concatenating a thought to a `SimplePath`. Since `pathToSibling` concatenates a thought to a `Path`, that guarantee doesn't exist here.

You can use `util/simplifyPath` for this purpose.

Reply:
Thanks! I've updated it